drivers/rc/lirc_dev.c

```diff
@@ -0,0 +1,1174 @@
+        break;
+
+      case LIRC_SET_REC_CARRIER_RANGE:
+        if (!lower->ops->s_rx_carrier_range)
+          {
+            ret = -ENOTTY;
+          }
+        else if (arg <= 0)
+          {
+            ret = -EINVAL;
+          }
+        else
+          {
+            fh->carrier_low = arg;
+          }
+        break;
+
+      case LIRC_GET_REC_RESOLUTION:
+        if (!lower->rx_resolution)
+          {
+            ret = -ENOTTY;
+          }
+        else
+          {
+            *val = lower->rx_resolution;
+          }
+        break;
+
+      case LIRC_SET_WIDEBAND_RECEIVER:
+        if (!lower->ops->s_learning_mode)
+          {
+            ret = -ENOTTY;
+          }
+        else
+          {
+            ret = lower->ops->s_learning_mode(lower, !!arg);
+          }
+        break;
+
+      case LIRC_SET_MEASURE_CARRIER_MODE:
+        if (!lower->ops->s_carrier_report)
+          {
+            ret = -ENOTTY;
+          }
+        else
+          {
+            ret = lower->ops->s_carrier_report(lower, !!arg);
+          }
+        break;
+
+      /* Generic timeout support */
+
+      case LIRC_GET_MIN_TIMEOUT:
+        if (!lower->min_timeout)
+          {
+            ret = -ENOTTY;
+          }
+        else
+          {
+            *val = lower->min_timeout;
+          }
+        break;
+
+      case LIRC_GET_MAX_TIMEOUT:
+        if (!lower->max_timeout)
+          {
+            ret = -ENOTTY;
+          }
+        else
+          {
+            *val = lower->max_timeout;
+          }
+        break;
+
+      case LIRC_SET_REC_TIMEOUT:
+        if (!lower->max_timeout)
+          {
+            ret = -ENOTTY;
+          }
+        else
+          {
+            if (arg < lower->min_timeout || arg > lower->max_timeout)
+              {
+                ret = -EINVAL;
+              }
+            else if (lower->ops->s_timeout)
+              {
+                ret = lower->ops->s_timeout(lower, arg);
+              }
+            else
+              {
+                lower->timeout = arg;
+              }
+          }
+        break;
+
+      case LIRC_GET_REC_TIMEOUT:
+        if (!lower->timeout)
+          {
+            ret = -ENOTTY;
+          }
+        else
+          {
+            *val = lower->timeout;
+          }
+        break;
+
+      case LIRC_SET_REC_TIMEOUT_REPORTS:
+        if (lower->ops->driver_type != LIRC_DRIVER_IR_RAW)
+          {
+            ret = -ENOTTY;
+          }
+        else
+          {
+            fh->send_timeout_reports = !!arg;
+          }
+        break;
+
+      default:
+        ret = -ENOTTY;
+    }
+
+  nxsem_post(&fh->exclsem);
+  return ret;
+}
+
+static ssize_t lirc_write_pulse(FAR struct file *filep,
+                                FAR const char *buffer, size_t buflen)
+{
+  FAR struct lirc_fh_s *fh = filep->f_priv;
+  FAR struct lirc_lowerhalf_s *lower = fh->lower;
+  size_t count;
+  int ret;
+
+  if (buflen < sizeof(unsigned) || buflen % sizeof(unsigned))
```

Comment:
please use sizeof(unsigned int) as you did on read_mode2()

Reply:
done